tests/test_tenant_workflow_service_boundary_check.py

```
@@ -0,0 +1,130 @@
+            ],
+        )
+
+        errors = tenant_workflow_service_boundary.validate_repo(repo_root)
+
+        self.assertEqual(errors, [])
+
+    def test_shell_wrapper_resolves_repo_root_from_another_working_directory(self):
+        outside_dir = Path(tempfile.mkdtemp(prefix="tenant-workflow-boundary-cwd-"))
+        self.addCleanup(shutil.rmtree, outside_dir, ignore_errors=True)
+
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)